### PR TITLE
EWPP-3623: Add Webtools Wtag service.

### PR DESCRIPTION
**Pull request by 22Alexandra**

### CODE REVIEW

File: oe_webtools.libraries.yml [hunk 1]
```diff
@@ -3,3 +3,8 @@ drupal.webtools-smartloader:
   version: 1.x
   js:
     https://europa.eu/webtools/load.js: { type: external, attributes: { defer: true } }
+
+oe_webtools.component-collect:
```
Comment (imanoleguskiza) [inline oe_webtools.libraries.yml:7]: We don't seem to be using this library?

File: modules/oe_webtools_wtag/src/Plugin/Field/FieldWidget/WtagWidget.php [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+   */
+  public static function isApplicable(FieldDefinitionInterface $field_definition) {
+    // This widget is only applicable to unlimited cardinality SKOS Concept
+    // reference fields that reference the DET vocabulary.
+    if ($field_definition->getFieldStorageDefinition()->getCardinality() !== FieldStorageDefinitionInterface::CARDINALITY_UNLIMITED) {
+      return FALSE;
+    }
+
+    $concept_schemes = $field_definition->getSetting('handler_settings')['concept_schemes'];
+    if (!in_array('http://data.europa.eu/uxp/det', $concept_schemes)) {
```
Comment (imanoleguskiza) [inline modules/oe_webtools_wtag/src/Plugin/Field/FieldWidget/WtagWidget.php:81]: Should we allow to use this widget for a field that is configured to use the DET vocabulary AND some other vocabulary? It seems weird as it essentially prevents the user from choosing anything but the det vocabulary

File: modules/oe_webtools_wtag/src/Element/Wtag.php [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+      'service' => 'wtag',
+      'target' => '#' . $element['#id'],
+      'title' => $element['#modal_title'],
+      'description' => $element['#modal_description'],
+    ];
+
+    $element['#wtag'] = Markup::create(Json::encode($wtag_json));
+    $element['#attached']['library'][] = 'oe_webtools/drupal.webtools-smartloader';
+    unset($element['#attributes']['data-autocomplete-path']);
+    $element['#attributes']['maxlength'] = '100000000';
```
Comment (imanoleguskiza) [inline modules/oe_webtools_wtag/src/Element/Wtag.php:66]: Are both maxlengths needed?

**Review by imanoleguskiza — CHANGES_REQUESTED** [review]

File: modules/oe_webtools_wtag/src/Plugin/Field/FieldWidget/WtagWidget.php (hunk 2 above)
Comment (22Alexandra) [reply modules/oe_webtools_wtag/src/Plugin/Field/FieldWidget/WtagWidget.php:81]: The Webtools Wtag widget only allows DET, so I don't think the user would be able to choose terms from other vocabularies.

File: oe_webtools.libraries.yml (hunk 1 above)
Comment (upchuk) [reply oe_webtools.libraries.yml:7]: @imanoleguskiza we will in EWCMS, and it's generic. Used in cases when we need to add aa webtools widget as part of an ajax request.

File: modules/oe_webtools_wtag/src/Plugin/Field/FieldWidget/WtagWidget.php (hunk 2 above)
Comment (upchuk) [reply modules/oe_webtools_wtag/src/Plugin/Field/FieldWidget/WtagWidget.php:81]: @imanoleguskiza They only built support for DET, quite hardcoded on their end as well..

File: modules/oe_webtools_wtag/src/Element/Wtag.php (hunk 3 above)
Comment (upchuk) [reply modules/oe_webtools_wtag/src/Element/Wtag.php:66]: Well, I think so, don't remember 100%. One is for the HTML element itself, the other is checked by a validator somewhere..

File: modules/oe_webtools_wtag/src/Plugin/Field/FieldWidget/WtagWidget.php (hunk 2 above)
Comment (upchuk) [reply modules/oe_webtools_wtag/src/Plugin/Field/FieldWidget/WtagWidget.php:81]: @imanoleguskiza Yes, true.
@22Alexandra could you please update this to ensure that there is a single concept scheme selected and that it is DET? Thanks

**Review by imanoleguskiza — APPROVED** [review]

**Review by imanoleguskiza — APPROVED** [review]

**Review by upchuk — APPROVED** [review]

**Pull request merged**